src/GraphODataTemplateWriter/CodeHelpers/Java/TypeHelperJava.cs

```diff
@@ -132,7 +133,53 @@ public static string TemplateName(this string templateFile)
         {
             return templateFile.Substring(templateFile.LastIndexOf("Templates"));
         }
+        private static Regex castPropsMatcher = new Regex(@"As[A-Z]");
+        private static Regex lastIdReplacer = new Regex(@"\/{id}$");
 
+        /// <summary>
+        /// Gets the canonical path of the entity for reference request
+        /// </summary>
+        /// <param name="c">Current navigation property class</param>
+        /// <param name="m">Current model</param>
+        /// <returns>The canonical version relative path of the entity for reference request</returns>
+        public static string CanonicalPath(this OdcmObject c, OdcmModel m)
+        {
+            var propertyInformation = GetCanonicalNavigationProperties(c, m, true); //singleton type all have ContainsTarget=True to support navigation properties bindings, but they don't actually contain target
+            var shortestCanonicalPath = propertyInformation
+                                        .Select(x => GetNavigationPathFromProperty(x, m))
+                                        .Select(x => new { path = x, slashCount = x.Count(y => y == '/') })
+                                        .OrderBy(x => x.slashCount)
+                                        .Select(x => x.path)
+                                        .FirstOrDefault();
+            return string.IsNullOrEmpty(shortestCanonicalPath) ? string.Empty : lastIdReplacer.Replace(shortestCanonicalPath, string.Empty);
+        }
+        public static bool ShouldIncludeReferenceUpdateMethods(this OdcmObject c, OdcmModel m)
+        {
+            return GetCanonicalNavigationProperties(c, m).Any();
+        }
+        private static IEnumerable<OdcmProperty> GetCanonicalNavigationProperties(this OdcmObject c, OdcmModel m, bool includeContainsTarget = false, bool includeNonCollectionProperties = false)
```

Comment:
Can you please add a summary for this method?

Reply:
added.